### DISCUSSION TITLE
bug report on opening  skeletalmesh at UE4.27?

### DISCUSSION THREAD

**Discussion by zgy001** [gildor2/UEViewer#252] [Show and tell]
Recently,i use UEViewer to open assets made by UE4.27,most of them can be opened correctly,but many skeletalmesh crashed.After review the code and compared with UE4.27 source code,i found that in 
UnMesh4.cpp line 1679 begin to SerializeStreamedData
at the end part,UEViewer serializes a tmp SkinWeightProfilesData and then process the data and return.But in
UE4.27 sourcecode in SkeletalMeshLODRenderData.cpp function FSkeletalMeshLODRenderData::SerializeStreamedData,
after serialize the last SkinWeightProfilesData,they then serialize a Tarray<uint8>  called RayTracingData and return.if i not add the Tarray<uint8> after serialize the tmp SkinWeightProfilesData,UEViewer may crash when serialize the second or third or else LOD.But when i add the Extra TAarry<uint8>,the crash disappeared and the mesh can be opened correctly!Is it a bug?thanks!

**Comment by gildor2** [comment]
What game was that? Some games are using customized data formats, so it's usual thing when something doesn't work.

**Reply by zgy001** [reply]
I try with the basic third person game,use the startercontent SK_Mannequin.uasset and SK_Mannequin_Female.uasset

**Comment by gildor2** [comment]
Fixed in https://github.com/gildor2/UEViewer/commit/31c047c8feb6cb87572c1fd7d855f9a88f8abcc4
Thanks for pointing out at issue!
